ironfish/src/rpc/routes/chain/getBlocks.ts

```diff
@@ -0,0 +1,132 @@
+  })
+  .defined()
+
+export const GetBlocksResponseSchema: yup.ObjectSchema<GetBlocksResponse> = yup
+  .object({
+    blocks: yup.array(RpcBlockSchema).defined(),
+  })
+  .defined()
+
+routes.register<typeof GetBlocksRequestSchema, GetBlocksResponse>(
+  `${ApiNamespace.chain}/getBlocks`,
+  GetBlocksRequestSchema,
+  async (request, context): Promise<void> => {
+    Assert.isInstanceOf(context, FullNode)
+
+    // Use negative numbers to start from the head of the chain
+    if (request.data.start <= 0) {
+      request.data.start = Math.max(request.data.start, GENESIS_BLOCK_SEQUENCE)
+    }
+
+    if (request.data.end - request.data.start > MAX_BLOCKS_RANGE) {
+      request.data.end = request.data.start + MAX_BLOCKS_RANGE
+    }
+
+    const blocks: RpcBlock[] = []
+    for (let seq = request.data.start; seq < request.data.end; seq++) {
+      const block = await getBlockWithSequence(context, seq)
+      blocks.push(block)
+    }
+
+    request.end({ blocks })
+  },
+)
+
+const getBlockWithSequence = async (node: FullNode, sequence: number): Promise<RpcBlock> => {
+  const header = await node.chain.getHeaderAtSequence(sequence)
+  let error = ''
+  if (!header) {
+    error = `No block found with sequence ${sequence}`
+    throw new RpcNotFoundError(error)
+  }
+
+  if (header.noteSize === null) {
+    throw new RpcValidationError('Block header was saved to database without a note size')
+  }
+
+  const block = await node.chain.getBlock(header)
+  if (!block) {
+    throw new RpcNotFoundError(`No block with header ${header.hash.toString('hex')}`)
+  }
+
+  const transactions: RpcTransaction[] = []
+
+  for (const tx of block.transactions) {
+    transactions.push({
+      hash: tx.hash().toString('hex'),
+      size: getTransactionSize(tx),
+      fee: Number(tx.fee()),
+      expiration: tx.expiration(),
+      notes: tx.notes.map((note) => ({
+        commitment: note.hash().toString('hex'),
+        hash: note.hash().toString('hex'),
+        serialized: note.serialize().toString('hex'),
+      })),
+      spends: tx.spends.map((spend) => ({
+        nullifier: spend.nullifier.toString('hex'),
+        commitment: spend.commitment.toString('hex'),
+        size: spend.size,
+      })),
+      mints: tx.mints.map((mint) => ({
+        id: mint.asset.id().toString('hex'),
+        metadata: BufferUtils.toHuman(mint.asset.metadata()),
+        name: BufferUtils.toHuman(mint.asset.name()),
+        creator: mint.asset.creator().toString('hex'),
+        value: mint.value.toString(),
+        transferOwnershipTo: mint.transferOwnershipTo?.toString('hex'),
+        assetId: mint.asset.id().toString('hex'),
+        assetName: mint.asset.name().toString('hex'),
+      })),
+      burns: tx.burns.map((burn) => ({
+        id: burn.assetId.toString('hex'),
+        value: burn.value.toString(),
+        assetId: burn.assetId.toString('hex'),
+        assetName: '',
+      })),
+      signature: tx.transactionSignature().toString('hex'),
+    })
```

Comment:
Could you refactor this out of `getBlock` and `getBlocks` into a function in `rpc/routes/chain/utils.ts` called `serializeRpcTransaction` that takes a `Transaction` and returns a `RpcTransaction`? Since we're using the same code in both places.